wallet/models.go

```diff
@@ -63,29 +63,30 @@ type GetBalanceResponse struct {
 
 // GetAddressRequest represents the request model for GetAddress
 type GetAddressRequest struct {
-	// Return subaddresses for this account.
-	AccountIndex uint64 `json:"account_index"`
-	// List of subaddresses to return from an account.
-	AddressIndex []uint64 `json:"address_index,omitempty"`
+	// Return addresses for this account.
+	AccountIndex uint32 `json:"account_index"`
+	// (Optional, defaults to all) List of address indices to return from the
+	// account. Index 0 is primary, 1+ are subadddresses.
+	AddressIndices []uint32 `json:"address_index,omitempty"`
 }
 
 // Address model
 type Address struct {
-	// The 95-character hex (sub)address string.
+	// The base58 (sub)address string.
 	Address string `json:"address"`
 	// Label of the (sub)address
 	Label string `json:"label"`
-	// index of the subaddress
-	AddressIndex uint64 `json:"address_index"`
+	// index of the (sub)address, 0 is primary
+	AddressIndex uint32 `json:"address_index"`
 	// states if the (sub)address has already received funds
 	Used bool `json:"used"`
 }
 
 // GetAddressResponse represents the response model for GetAddress
 type GetAddressResponse struct {
-	// The 95-character hex address string of the monero-wallet-rpc in session.
+	// The primary address of the requested account index.
 	Address string `json:"address"`
-	// array of addresses informations
+	// Array of address information entries
 	Addresses []Address `json:"addresses"`
```

Comment:
I also opened a pull request to update the official documentation:
https://github.com/monero-project/monero-site/pull/2186